tutorials/en/llama3-with-ollama.mdx

```diff
@@ -0,0 +1,185 @@
+    'role': 'user',
+    'content': prompt,
+},
+])
+print(response['message']['content'])
+```
+
+### Streaming Responses
+For applications requiring real-time feedback, you can enable streaming responses. This allows you to receive parts of the response as they are generated:
+
+```python
+import ollama
+
+ollama_response = ollama.chat(
+    model='llama3',
+    stream=True,
+    messages=[
+        {
+            'role': 'user',
+            'content': 'Explain the Big Bang theory',
+        },
+    ]
+)
+
+for chunk in ollama_response:
+    print(chunk['message']['content'], end='', flush=True)
+```
+
+### Ongoing Dialogue with Context
+Maintaining context in a conversation allows for more natural interactions. Here's how you can manage ongoing dialogue:
+
+```python
+import ollama
+
+chat_messages = []
+
+def create_message(content, role):
+    return {'role': role, 'content': content}
+
+def chat():
+    response = ollama.chat(model='llama3', stream=True, messages=chat_messages)
+    assistant_message = ''
+    for chunk in response:
+        assistant_message += chunk['message']['content']
+        print(chunk['message']['content'], end='', flush=True)
+    chat_messages.append(create_message(assistant_message, 'assistant'))
+
+def ask(message):
+    chat_messages.append(create_message(message, 'user'))
+    print(f'\n\n--{message}--\n\n')
+    chat()
+
+ask('What is the capital of France?')
+ask('How far is it from Paris to Berlin?')
+```
+
+### Text Completion
+You can use Ollama with **Llama3** for text completion tasks, such as code generation or completing sentences by using the `generate` function:
+
+```python
+import ollama
+
+response = ollama.generate(
+    model='llama3',
+    prompt='Once upon a time, in a faraway land,'
+)
+print(response['response'])
+```
+
+### Custom Clients
+You can also create custom client to interact with the Ollama server. Here's an example of a custom client:
+
+```python
+from ollama import Client
+
+client = Client(host='http://your-ollama-server')
+response = client.chat(model='llama3', messages=[
+    {
+        'role': 'user',
+        'content': 'What is machine learning?',
+    }
+])
+print(response['message']['content'])
+```
+
+### Generating SQL from Text
+Ollama can be used to generate SQL queries from natural language inputs. Here's how to set up a local instance and use it:
+
+```python
+from ollama import Client
+
+client = Client(host='http://localhost:11434')
+
+prompt = 'Get the top 10 most expensive products.'
+response = client.chat(model='llama3', messages=[
+    {
+        'role': 'system',
+        'content': """Here is the database schema that the SQL query will run on:
+    CREATE TABLE products (
+      product_id INTEGER PRIMARY KEY,
+      name VARCHAR(50),
+      price DECIMAL(10,2),
+      quantity INTEGER
+    );
+    """,
+    },
+    {
+        'role': 'user',
+        'content': prompt,
+    }
+])
+print(response['message']['content'])
+```
+

```

Comment:
I wudl add all these implementations first and then an eample (see comment above() like this is an example implementatio for X task, and you can run the model running the comand 
```
python3 llama3_demo.py
```
blabla in the terminal